google/cloud/iam/samples/iam_samples.cc

```diff
@@ -603,7 +603,8 @@ void AutoRun(std::vector<std::string> const& argv) {
       GetServiceAccountKey({sample_service_account_key_name});
     } catch (std::runtime_error const&) {
       // Service Account Key may not be usable for up to 60s after creation.
-      std::this_thread::sleep_for(std::chrono::seconds(61));
+      // Plus some unknown amount of time for propagation.
```

Comment:
Consider something like:

```cc
for (auto backoff : {65, 30, 30, 30, 30, 0}) {
  try {
    GetServiceAccountKey({sample_service_account_key_name});
    break;
  } catch (std::runtime_error const&) {
    if (backoff == 0) throw; // give up and raise the error
    std::this_thread:sleep_for(std::chrono::seconds(backoff));
  }
}
```

Reply:
Should we look for `NOT_FOUND` explicitly?